compiler-rt/test/nsan/nan.cpp

```diff
@@ -0,0 +1,19 @@
+// RUN: %clangxx_nsan -O0 -g %s -o %t
```

Comment:
We also need to add a run for -O3.
See https://stackoverflow.com/questions/54481855/clang-ignoring-attribute-noinline if optimizations trigger constant folding here.
I'd recommend a slight modification (probably should suffice to make things work):
```
__attribute__((noinline)) float f(float p, float q) {
     float  ret = p / q;
     return ret;
}
```
And then call f(0., 0.)